src/shogun/base/subset_iterators.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Gil Hoben
+ */
+
+#ifndef SHOGUN_SUBSET_ITERATORS_H
+#define SHOGUN_SUBSET_ITERATORS_H
+
+#include <shogun/labels/Labels.h>
+#include <shogun/features/Features.h>
+
+namespace shogun
+{
+	template <
+	    typename T, std::enable_if<
+	                    (std::is_base_of<T, CFeatures>::value) ||
+	                    (std::is_base_of<T, CLabels>::value)>* = nullptr>
+	class SubsetIterator
```

Comment:
Find the name a bit Strang but I don’t know a better one :)

Reply:
hmm i agree, i just thought it was easier to distinguish from the vector iterator